firmware/quadruna/BMS/boot/cubemx/Src/main.c

```diff
@@ -105,7 +108,29 @@ void        runCanTxTask(void *argument);
 
 /* Private user code ---------------------------------------------------------*/
 /* USER CODE BEGIN 0 */
-CanHandle can = { .can = &hfdcan1, .can_msg_received_callback = io_can_pushRxMsgToQueue };
+CanHandle can = { .hcan = &hfdcan1 };
+
+void HAL_FDCAN_RxFifo0Callback(FDCAN_HandleTypeDef *hfdcan, const uint32_t RxFifo0ITs)
+{
+    assert(hfdcan == &hfdcan1);
+    UNUSED(RxFifo0ITs);
+    CanMsg rx_msg;
+    if (!io_can_receive(&can, FDCAN_RX_FIFO0, &rx_msg))
+        // Early return if RX msg is unavailable.
+        return;
+    io_canQueue_pushRx(&rx_msg);
+}
+
+void HAL_FDCAN_RxFifo1Callback(FDCAN_HandleTypeDef *hfdcan, const uint32_t RxFifo1ITs)
+{
+    assert(hfdcan == &hfdcan1);
+    UNUSED(RxFifo1ITs);
+    CanMsg rx_msg;
+    if (!io_can_receive(&can, FDCAN_RX_FIFO1, &rx_msg))
+        // Early return if RX msg is unavailable.
+        return;
+    io_canQueue_pushRx(&rx_msg);
+}
```

Comment:
Any reason you have the callbacks in `main.c` for FDCAN? They're all in `hw_can.c` for regular CAN

Reply:
i was doing a lot of thinking about this. I think there are a lot of good reasons to do both, but I think we should revert to old because of a recent change where canqueues are global (despite multiple busses). My understanding is that these two callbacks it regardless of how many CAN busses we are using?

Reply:
Yeah I think these callbacks are always fired. And then you can check which `hfdcan` you get to check which bus its from?

Reply:
I just copy pasted and I remembered that the shared code cannot check which canhandle is active because it differs by board. Otherwise, we would have to handle multibus behaviour in (global) canQueue? seems like bad seperation of concerns.